.github/workflows/deploy_api.yml

```diff
@@ -9,24 +9,131 @@ on:
   schedule:
    - cron: '30 0,12 * * *'
 
+  # push:
   # Hook to trigger a manual run.
   # See: https://goobar.io/2019/12/07/manually-trigger-a-github-actions-workflow/
   repository_dispatch:
     types: publish-api
 
 jobs:
-  publish-api:
+  execute-model:
     runs-on: self-hosted
+    env:
+      # To pin to an old data set or model code, put the branch/tag/commit here:
+      COVID_DATA_PUBLIC_REF: 'master'
+      # !!! Change this to your BRANCH if you want to test it
+      COVID_DATA_MODEL_REF: 'master'
+      SNAPSHOT_ID: ${{github.run_number}}
+    steps:
+    - name: Checkout covid-data-model
+      uses: actions/checkout@v2
+      with:
+        repository: covid-projections/covid-data-model
+        path: covid-data-model
+        ref: '${{env.COVID_DATA_MODEL_REF}}'
+    - name: Checkout covid-data-public
+      uses: actions/checkout@v2
+      with:
+        repository: covid-projections/covid-data-public
+        path: covid-data-public
+        lfs: true
+        ref: '${{env.COVID_DATA_PUBLIC_REF}}'
+    - name: Setup Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: '3.7.6'
+        architecture: 'x64'
+    - name: Cache Pip
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip-${{ hashFiles('requirements.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
+          ${{ runner.os }}-
+    - name: Install Dependencies
+      working-directory: ./covid-data-model
+      run: pip install -r requirements.txt
 
+    # - name: create temp file
+    #   run: |
+    #     mkdir -p /data/api-results-${{env.SNAPSHOT_ID}}
+    #     touch /data/api-results-${{env.SNAPSHOT_ID}}/deleteme
+    - name: Build Model Results (run.sh .. .. execute_model)
+      env:
+        COVID_MODEL_CORES: 96
+      run: ./covid-data-model/run.sh ./covid-data-public /data/api-results-${{env.SNAPSHOT_ID}} execute_model
```

Comment:
I think this code might be cleaner with some more environment variables!
```

env: 
  COVID_DATA_PUBLIC: ./covid-data-public
run: ./covid-data-model/run.sh ${COVID_DATA_PUBLIC}
```

Also, you can add top-level environment variables in the workflow if that's helpful!

Reply:
I'm not sure i follow.